### PR TITLE
Improved: BatchModal component with added - swipe left to right to skip the job functionality

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

#2d92ghd

Closes #
https://github.com/hotwax/job-manager/commit/6d55fb97c65abaef33dae2fefd34c9cb0dd9b7d1

### Short Description and Why It's Useful

Now user will easily be able to skip the job by swiping left to right and confirming to skip it.

### Screenshots of Visual Changes before/after (If There Are Any)

https://user-images.githubusercontent.com/62360781/169819218-e3db7f51-be1c-47a0-b2bb-14685a05cd9d.mov

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/views/Orders.vue [hunk 1]
```diff
@@ -146,6 +151,8 @@ import {
   modalController
 } from '@ionic/vue';
 import { defineComponent } from 'vue';
+import { showToast } from "@/utils";
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:154]: Move this import of `showToast` to the previous import from `@/utils`

File: src/store/modules/job/actions.ts [hunk 2]
```diff
@@ -453,7 +453,8 @@ const actions: ActionTree<JobState, RootState> = {
   async skipJob({ commit, getters }, job) {
     let skipTime = {};
     const integer1 = getters['getTemporalExpr'](job.tempExprId)?.integer1;
-    const integer2 = getters['getTemporalExpr'](job.tempExprId)?.integer2
+    const integer2 = getters['getTemporalExpr'](job.tempExprId)?.integer2;
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:456]: Remove this unwanted change.

File: src/views/Orders.vue [hunk 3]
```diff
@@ -102,6 +102,11 @@
                 <ion-label class="ion-text-wrap">{{ batch?.jobName }}</ion-label>
                 <ion-note slot="end">{{ batch?.runTime ? getTime(batch.runTime) : '' }}</ion-note>
               </ion-item>
+              <ion-item-options side="start">
+                <ion-item-option @click="skipBatch(batch)" color="secondary" style="color:black">
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:106]: I think we don't need any style property here.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Orders.vue (hunk 3 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:106]: Done sir, removed!

File: src/views/Orders.vue (hunk 1 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:154]: Okay sir! done.

File: src/views/Orders.vue [hunk 4]
```diff
@@ -246,6 +252,28 @@ export default defineComponent({
...
+          buttons: [
+            {
+              text: this.$t("Don't skip"),
+              role: 'cancel',
+            },
+            {
+              text: this.$t('Skip'),
+              handler: async () => {
+                this.store.dispatch('job/skipJob', batch).then(() => {
+                  showToast(translate("This job has been skipped"))
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:269]: Improve the condition to display the toast only when we have success in response, currently, this message will always be displayed even if the API returns some error.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Orders.vue (hunk 4 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:269]: Sure sir! Working on it.

File: src/views/Orders.vue [hunk 5]
```diff
@@ -246,6 +252,34 @@ export default defineComponent({
...
+          buttons: [
+            {
+              text: this.$t("Don't skip"),
+              role: 'cancel',
+            },
+            {
+              text: this.$t('Skip'),
+              handler: async () => {
+                this.store.dispatch('job/skipJob', batch).then((res) => {
+                  if (res.status === 200) {
```
Comment (adityasharma7) [inline src/views/Orders.vue:269]: Handle error properly 
https://github.com/hotwax/job-manager/blob/main/src/store/modules/job/actions.ts#L478

File: src/views/Orders.vue [hunk 6]
```diff
@@ -246,6 +252,34 @@ export default defineComponent({
...
+                    showToast(translate("This job has been skipped"));
+                  }
+                  else {
```
Comment (adityasharma7) [inline src/views/Orders.vue:272]: If-else should be in continuity

File: src/views/Orders.vue [hunk 7]
```diff
@@ -246,6 +252,34 @@ export default defineComponent({
...
+                    showToast(translate("This job schedule cannot be skipped"));
+                    return;
```
Comment (adityasharma7) [inline src/views/Orders.vue:274]: Why this return is needed?

File: src/views/Orders.vue [hunk 8]
```diff
@@ -246,6 +252,34 @@ export default defineComponent({
         });
       return deleteBatchAlert.present();
     },
+    async skipBatch (batch: any) {
+      const alert = await alertController
```
Comment (adityasharma7) [inline src/views/Orders.vue:256]:
```suggestion
      const skipJobAlert = await alertController
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Orders.vue (hunk 5 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:269]: Yes sir!

File: src/views/Orders.vue (hunk 6 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:272]: Sure sir! Woking on it.

File: src/views/Orders.vue (hunk 8 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:256]: Done sir! fixed it.

File: src/views/Orders.vue (hunk 7 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:274]: Got it sir! There is no need of return. It should simply show the unsuccessful toast.

**Review by ymaheshwari1 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**